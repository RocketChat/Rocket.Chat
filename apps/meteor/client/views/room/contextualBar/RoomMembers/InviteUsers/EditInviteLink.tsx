import type { SelectOption } from '@rocket.chat/fuselage';
import { Box, Field, FieldLabel, FieldRow, Select, Button } from '@rocket.chat/fuselage';
import type { ReactElement } from 'react';
import { useMemo } from 'react';
import { useForm, Controller } from 'react-hook-form';
import { useTranslation } from 'react-i18next';

type EditInviteLinkProps = {
	daysAndMaxUses: { days: string; maxUses: string };
	onClickNewLink: (daysAndMaxUses: { days: string; maxUses: string }) => void;
};

const EditInviteLink = ({ daysAndMaxUses, onClickNewLink }: EditInviteLinkProps): ReactElement => {
	const { t } = useTranslation();
	const {
		handleSubmit,
		formState: { isDirty, isSubmitting },
		control,
	} = useForm({ defaultValues: { days: daysAndMaxUses.days, maxUses: daysAndMaxUses.maxUses } });

	const daysOptions: SelectOption[] = useMemo(
		() => [
			['1', '1'],
			['7', '7'],
			['15', '15'],
			['30', '30'],
			['0', t('Never')],
		],
		[t],
	);

	const maxUsesOptions: SelectOption[] = useMemo(
		() => [
			['5', '5'],
			['10', '10'],
			['25', '25'],
			['50', '50'],
			['100', '100'],
			['0', t('No_Limit')],
		],
		[t],
	);

	return (
		<>
			<Field>
				<FieldLabel flexGrow={0}>{t('Expiration_(Days)')}</FieldLabel>
				<FieldRow>
					<Controller
						name='days'
						control={control}
						render={({ field: { onChange, value, name } }): ReactElement => (
							<Select name={name} value={value} onChange={onChange} options={daysOptions} />
						)}
					/>
				</FieldRow>
			</Field>
			<Field>
				<FieldLabel flexGrow={0}>{t('Max_number_of_uses')}</FieldLabel>
				<FieldRow>
					<Controller
						name='maxUses'
						control={control}
						render={({ field: { onChange, value, name } }): ReactElement => (
							<Select name={name} value={value} onChange={onChange} options={maxUsesOptions} />
						)}
					/>
				</FieldRow>
			</Field>
			<Box mbs={8}>
				<Button loading={isSubmitting} disabled={!isDirty} primary onClick={handleSubmit(onClickNewLink)}>
					{t('Generate_New_Link')}
				</Button>
			</Box>
		</>
	);
};

export default EditInviteLink;
