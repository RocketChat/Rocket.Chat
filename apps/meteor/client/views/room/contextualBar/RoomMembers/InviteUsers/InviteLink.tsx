import { Box, Field, FieldLabel, FieldRow, UrlInput, Icon, Button, InputBox } from '@rocket.chat/fuselage';
import type { ReactElement } from 'react';
import { useTranslation } from 'react-i18next';

import useClipboardWithToast from '../../../../../hooks/useClipboardWithToast';

type InviteLinkProps = {
	linkText: string;
	captionText: string;
	onClickEdit: () => void;
};

const InviteLink = ({ linkText, captionText, onClickEdit }: InviteLinkProps): ReactElement => {
	const { t } = useTranslation();
	const { copy } = useClipboardWithToast(linkText);

	return (
		<>
			<Field>
				<FieldLabel flexGrow={0}>{t('Invite_Link')}</FieldLabel>
				<FieldRow>
					{!linkText && <InputBox.Skeleton />}
					{linkText && <UrlInput value={linkText} addon={<Icon onClick={(): Promise<void> => copy()} name='copy' size='x16' />} />}
				</FieldRow>
				{captionText && (
					<Box pb={8} color='annotation' fontScale='c2'>
						{captionText}
					</Box>
				)}
			</Field>
			{onClickEdit && (
				<Box mbs={8}>
					<Button onClick={onClickEdit}>{t('Edit_Invite')}</Button>
				</Box>
			)}
		</>
	);
};

export default InviteLink;
