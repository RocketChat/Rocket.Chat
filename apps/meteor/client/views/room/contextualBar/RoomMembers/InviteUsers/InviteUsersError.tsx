import { Callout } from '@rocket.chat/fuselage';
import type { ReactElement } from 'react';

import InviteUsersWrapper from './InviteUsersWrapper';

type InviteUsersProps = {
	onClose: () => void;
	error: Error;
	onClickBack?: (() => void) | undefined;
};

const InviteUsersError = ({ onClose, error, onClickBack }: InviteUsersProps): ReactElement => (
	<InviteUsersWrapper onClose={onClose} onClickBack={onClickBack}>
		<Callout type='danger'>{(error || '').toString()}</Callout>
	</InviteUsersWrapper>
);

export default InviteUsersError;
