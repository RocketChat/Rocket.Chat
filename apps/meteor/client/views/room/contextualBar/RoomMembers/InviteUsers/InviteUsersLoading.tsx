import { Skeleton } from '@rocket.chat/fuselage';
import type { ReactElement } from 'react';

import InviteUsersWrapper from './InviteUsersWrapper';

type InviteUsersProps = {
	onClose: () => void;
	onClickBack: (() => void) | undefined;
};

const InviteUsersLoading = ({ onClose, onClickBack: onClickBackMembers }: InviteUsersProps): ReactElement => (
	<InviteUsersWrapper onClose={onClose} onClickBack={onClickBackMembers}>
		<Skeleton w='full' />
	</InviteUsersWrapper>
);

export default InviteUsersLoading;
