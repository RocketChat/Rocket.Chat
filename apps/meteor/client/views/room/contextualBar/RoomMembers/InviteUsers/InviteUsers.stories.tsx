import type { Meta, StoryFn } from '@storybook/react';

import InviteUsers from './InviteUsers';
import { Contextualbar } from '../../../../../components/Contextualbar';

export default {
	title: 'Room/Contextual Bar/RoomMembers/InviteUsers',
	component: InviteUsers,
	parameters: {
		layout: 'fullscreen',
		actions: { argTypesRegex: '^on.*' },
	},
	decorators: [(fn) => <Contextualbar height='100vh'>{fn()}</Contextualbar>],
} satisfies Meta<typeof InviteUsers>;

export const Default: StoryFn<typeof InviteUsers> = (args) => <InviteUsers {...args} />;
Default.storyName = 'InviteUsers';
Default.args = {
	linkText: 'https://go.rocket.chat/invite?host=open.rocket.chat&path=invite%2F5sBs3a',
	captionText: 'Expire on February 4, 2020 4:45 PM.',
};
