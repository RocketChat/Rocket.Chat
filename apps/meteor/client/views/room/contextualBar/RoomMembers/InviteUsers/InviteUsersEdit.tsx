import type { ReactElement } from 'react';

import EditInviteLink from './EditInviteLink';
import InviteUsersWrapper from './InviteUsersWrapper';

type InviteUsersEditProps = {
	onClickBackLink?: () => void;
	onClickNewLink: (daysAndMaxUses: { days: string; maxUses: string }) => void;
	onClose: () => void;
	daysAndMaxUses: { days: string; maxUses: string };
};

const InviteUsersEdit = ({ onClickBackLink, onClickNewLink, onClose, daysAndMaxUses }: InviteUsersEditProps): ReactElement => {
	return (
		<InviteUsersWrapper onClickBack={onClickBackLink} onClose={onClose}>
			<EditInviteLink onClickNewLink={onClickNewLink} daysAndMaxUses={daysAndMaxUses} />
		</InviteUsersWrapper>
	);
};

export default InviteUsersEdit;
