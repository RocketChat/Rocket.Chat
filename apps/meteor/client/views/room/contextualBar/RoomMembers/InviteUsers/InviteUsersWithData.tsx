import type { IRoom } from '@rocket.chat/core-typings';
import { useEffectEvent } from '@rocket.chat/fuselage-hooks';
import { useEndpoint, useTranslation, useToastMessageDispatch } from '@rocket.chat/ui-contexts';
import type { ReactElement } from 'react';
import { useState, useEffect } from 'react';

import InviteUsers from './InviteUsers';
import { useFormatDateAndTime } from '../../../../../hooks/useFormatDateAndTime';
import { useRoomToolbox } from '../../../contexts/RoomToolboxContext';

type InviteUsersWithDataProps = {
	rid: IRoom['_id'];
	onClickBack: () => void;
};

const InviteUsersWithData = ({ rid, onClickBack }: InviteUsersWithDataProps): ReactElement => {
	const t = useTranslation();
	const dispatchToastMessage = useToastMessageDispatch();
	const [
		{
			isEditing,
			url,
			caption,
			error,
			daysAndMaxUses: { days, maxUses },
		},
		setInviteState,
	] = useState({
		isEditing: false,
		daysAndMaxUses: { days: '1', maxUses: '0' },
		url: '',
		caption: '',
		error: undefined as Error | undefined,
	});

	const { closeTab } = useRoomToolbox();
	const format = useFormatDateAndTime();
	const findOrCreateInvite = useEndpoint('POST', '/v1/findOrCreateInvite');

	const handleEdit = useEffectEvent(() => setInviteState((prevState) => ({ ...prevState, isEditing: true })));
	const handleBackToLink = useEffectEvent(() => setInviteState((prevState) => ({ ...prevState, isEditing: false })));

	const linkExpirationText = useEffectEvent(
		(data: {
			days: number;
			maxUses: number;
			rid: string;
			userId: string;
			createdAt: string;
			expires: string | null;
			uses: number;
			url: string;
			_id: string;
			_updatedAt: string;
		}) => {
			if (!data) {
				return '';
			}

			if (data.expires) {
				const expiration = new Date(data.expires);
				if (data.maxUses) {
					const usesLeft = data.maxUses - data.uses;
					return t('Your_invite_link_will_expire_on__date__or_after__usesLeft__uses', {
						date: format(expiration),
						usesLeft,
					});
				}

				return t('Your_invite_link_will_expire_on__date__', { date: format(expiration) });
			}

			if (data.maxUses) {
				const usesLeft = data.maxUses - data.uses;
				return t('Your_invite_link_will_expire_after__usesLeft__uses', { usesLeft });
			}

			return t('Your_invite_link_will_never_expire');
		},
	);

	useEffect(() => {
		(async (): Promise<void> => {
			try {
				const data = await findOrCreateInvite({ rid, days: Number(days), maxUses: Number(maxUses) });
				setInviteState((prevState) => ({ ...prevState, url: data?.url, caption: linkExpirationText(data) }));
				dispatchToastMessage({ type: 'success', message: t('Invite_link_generated') });
			} catch (error) {
				setInviteState((prevState) => ({ ...prevState, error: error as Error }));
			}
		})();
	}, [dispatchToastMessage, t, findOrCreateInvite, linkExpirationText, rid, days, maxUses]);

	const handleGenerateLink = useEffectEvent((daysAndMaxUses: { days: string; maxUses: string }) => {
		setInviteState((prevState) => ({ ...prevState, daysAndMaxUses, isEditing: false }));
	});

	return (
		<InviteUsers
			isEditing={isEditing}
			error={error}
			linkText={url}
			captionText={caption}
			daysAndMaxUses={{ days, maxUses }}
			onClose={closeTab}
			onClickBackMembers={onClickBack}
			onClickBackLink={handleBackToLink}
			onClickEdit={handleEdit}
			onClickNewLink={handleGenerateLink}
		/>
	);
};

export default InviteUsersWithData;
