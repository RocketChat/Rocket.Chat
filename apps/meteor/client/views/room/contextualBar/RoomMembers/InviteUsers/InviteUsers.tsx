import { Callout } from '@rocket.chat/fuselage';
import type { ReactElement } from 'react';
import { useTranslation } from 'react-i18next';

import EditInviteLink from './EditInviteLink';
import InviteLink from './InviteLink';
import {
	ContextualbarHeader,
	ContextualbarTitle,
	ContextualbarBack,
	ContextualbarClose,
	ContextualbarScrollableContent,
} from '../../../../../components/Contextualbar';

type InviteUsersProps = {
	onClickBackMembers?: () => void;
	onClickBackLink?: () => void;
	onClickNewLink: (daysAndMaxUses: { days: string; maxUses: string }) => void;
	onClose: () => void;
	isEditing: boolean;
	daysAndMaxUses: { days: string; maxUses: string };
	onClickEdit: () => void;
	captionText: string;
	linkText: string;
	error?: Error;
};

const InviteUsers = ({
	onClickBackMembers,
	onClickBackLink,
	onClickNewLink,
	onClose,
	isEditing,
	onClickEdit,
	daysAndMaxUses,
	captionText,
	linkText,
	error,
}: InviteUsersProps): ReactElement => {
	const { t } = useTranslation();

	return (
		<>
			<ContextualbarHeader>
				{(onClickBackMembers || onClickBackLink) && <ContextualbarBack onClick={isEditing ? onClickBackLink : onClickBackMembers} />}
				<ContextualbarTitle>{t('Invite_Users')}</ContextualbarTitle>
				{onClose && <ContextualbarClose onClick={onClose} />}
			</ContextualbarHeader>
			<ContextualbarScrollableContent>
				{error && <Callout type='danger'>{error.toString()}</Callout>}
				{isEditing && !error && <EditInviteLink onClickNewLink={onClickNewLink} daysAndMaxUses={daysAndMaxUses} />}
				{!isEditing && !error && <InviteLink captionText={captionText} onClickEdit={onClickEdit} linkText={linkText} />}
			</ContextualbarScrollableContent>
		</>
	);
};

export default InviteUsers;
