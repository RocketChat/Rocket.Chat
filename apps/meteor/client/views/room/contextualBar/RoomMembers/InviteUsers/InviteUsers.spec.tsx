/* eslint-disable @typescript-eslint/naming-convention */
import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import * as stories from './InviteUsers.stories';

const testCases = Object.values(composeStories(stories)).map((Story) => [Story.storyName || 'Story', Story]);

beforeEach(() => {
	jest.mock('react', () => ({
		...jest.requireActual('react'),
		useId: () => 1,
	}));
});

test.each(testCases)(`renders %s without crashing`, async (_storyname, Story) => {
	const view = render(<Story />);
	expect(view.baseElement).toMatchSnapshot();
});

test.each(testCases)('%s should have no a11y violations', async (_storyname, Story) => {
	const { container } = render(<Story />);

	/**
	 ** Disable 'nested-interactive' rule because our `Select` component is still not a11y compliant
	 **/
	const results = await axe(container, { rules: { 'nested-interactive': { enabled: false } } });
	expect(results).toHaveNoViolations();
});
