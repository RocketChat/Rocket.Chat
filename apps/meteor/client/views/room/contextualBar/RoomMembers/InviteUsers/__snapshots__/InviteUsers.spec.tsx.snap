// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Default without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-vertical-bar rcx-css-ucmcg1"
    >
      <div
        class="rcx-box rcx-box--full rcx-css-ftwpdg"
      >
        <div
          class="rcx-box rcx-box--full rcx-css-1sl6k6j"
        >
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button rcx-css-x7bl3q rcx-css-1yzvz7u"
            title="Back"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-arrow-back rcx-icon rcx-css-4pvxx3"
            >
              ⦉
            </i>
          </button>
          <div
            class="rcx-box rcx-box--full rcx-css-x7bl3q rcx-css-1to6ka7"
            id="contextualbarTitle"
          >
            Invite users
          </div>
          <button
            aria-label="Close"
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button rcx-css-x7bl3q rcx-css-1yzvz7u"
            data-qa="ContextualbarActionClose"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
        </div>
      </div>
      <div
        class="rcx-box rcx-box--full rcx-css-1svuzur"
      >
        <div
          class="rcx-box rcx-box--full rcx-css-pln26h rcx-css-1cb6i7s"
          data-overlayscrollbars="host"
        >
          <div
            class="os-size-observer"
          >
            <div
              class="os-size-observer-listener"
            />
          </div>
          <div
            class=""
            data-overlayscrollbars-padding=""
            style="margin-right: 0px; margin-bottom: 0px; margin-left: 0px; top: 0px; left: 0px; width: calc(100% + 0px);"
          >
            <div
              class=""
              data-overlayscrollbars-viewport="overflowXHidden overflowYHidden"
              style="padding: 0px 0px 0px 0px;"
              tabindex="-1"
            >
              <div
                class="rcx-box rcx-box--full rcx-css-iag4sp"
              >
                <div
                  class="rcx-box rcx-box--full rcx-field rcx-css-1fuhgt0"
                >
                  <label
                    class="rcx-box rcx-box--full rcx-label rcx-box rcx-box--full rcx-field__label rcx-css-irxdij"
                    for=":r0:"
                  >
                    Invite link
                  </label>
                  <span
                    class="rcx-box rcx-box--full rcx-field__row"
                  >
                    <label
                      class="rcx-box rcx-box--full rcx-label rcx-box rcx-box--full rcx-box--animated rcx-input-box__wrapper"
                    >
                      <input
                        class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--undecorated rcx-input-box--type-url rcx-input-box"
                        id=":r0:"
                        size="1"
                        type="url"
                        value="https://go.rocket.chat/invite?host=open.rocket.chat&path=invite%2F5sBs3a"
                      />
                      <span
                        class="rcx-box rcx-box--full rcx-input-box__addon"
                      >
                        <i
                          aria-hidden="true"
                          class="rcx-box rcx-box--full rcx-icon--name-copy rcx-icon rcx-css-1g87xs3"
                        >
                          
                        </i>
                      </span>
                    </label>
                  </span>
                  <div
                    class="rcx-box rcx-box--full rcx-css-ox4caw"
                  >
                    Expire on February 4, 2020 4:45 PM.
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="os-scrollbar os-scrollbar-horizontal os-theme-dark os-scrollbar-auto-hide os-scrollbar-auto-hide-hidden os-scrollbar-handle-interactive os-scrollbar-cornerless os-scrollbar-unusable"
            style="--os-scroll-percent: 0; --os-viewport-percent: 0; --os-scroll-direction: 0;"
          >
            <div
              class="os-scrollbar-track"
            >
              <div
                class="os-scrollbar-handle"
              />
            </div>
          </div>
          <div
            class="os-scrollbar os-scrollbar-vertical os-theme-dark os-scrollbar-auto-hide os-scrollbar-auto-hide-hidden os-scrollbar-handle-interactive os-scrollbar-cornerless os-scrollbar-unusable"
            style="--os-scroll-percent: 0; --os-viewport-percent: 0; --os-scroll-direction: 0;"
          >
            <div
              class="os-scrollbar-track"
            >
              <div
                class="os-scrollbar-handle"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders InviteEdit without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-vertical-bar rcx-css-ucmcg1"
    >
      <div
        class="rcx-box rcx-box--full rcx-css-ftwpdg"
      >
        <div
          class="rcx-box rcx-box--full rcx-css-1sl6k6j"
        >
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button rcx-css-x7bl3q rcx-css-1yzvz7u"
            title="Back"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-arrow-back rcx-icon rcx-css-4pvxx3"
            >
              ⦉
            </i>
          </button>
          <div
            class="rcx-box rcx-box--full rcx-css-x7bl3q rcx-css-1to6ka7"
            id="contextualbarTitle"
          >
            Invite users
          </div>
          <button
            aria-label="Close"
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button rcx-css-x7bl3q rcx-css-1yzvz7u"
            data-qa="ContextualbarActionClose"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
        </div>
      </div>
      <div
        class="rcx-box rcx-box--full rcx-css-1svuzur"
      >
        <div
          class="rcx-box rcx-box--full rcx-css-pln26h rcx-css-1cb6i7s"
          data-overlayscrollbars="host"
        >
          <div
            class="os-size-observer"
          >
            <div
              class="os-size-observer-listener"
            />
          </div>
          <div
            class=""
            data-overlayscrollbars-padding=""
            style="margin-right: 0px; margin-bottom: 0px; margin-left: 0px; top: 0px; left: 0px; width: calc(100% + 0px);"
          >
            <div
              class=""
              data-overlayscrollbars-viewport="overflowXHidden overflowYHidden"
              style="padding: 0px 0px 0px 0px;"
              tabindex="-1"
            >
              <div
                class="rcx-box rcx-box--full rcx-css-iag4sp"
              >
                <div
                  class="rcx-box rcx-box--full rcx-field rcx-css-1fuhgt0"
                >
                  <label
                    class="rcx-box rcx-box--full rcx-label rcx-box rcx-box--full rcx-field__label rcx-css-irxdij"
                    for=":r1:"
                  >
                    Expiration (Days)
                  </label>
                  <span
                    class="rcx-box rcx-box--full rcx-field__row"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-labelledby="react-aria-:ra:"
                      class="rcx-box rcx-box--full rcx-select rcx-css-1vw6rc6"
                      id=":r1:"
                      type="button"
                    >
                      <div
                        aria-hidden="true"
                        data-a11y-ignore="aria-hidden-focus"
                        data-react-aria-prevent-focus="true"
                        data-testid="hidden-select-container"
                        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
                      >
                        <label>
                          <select
                            name="days"
                            tabindex="-1"
                          >
                            <option />
                            <option
                              value="1"
                            >
                              1
                            </option>
                            <option
                              value="7"
                            >
                              7
                            </option>
                            <option
                              value="15"
                            >
                              15
                            </option>
                            <option
                              value="30"
                            >
                              30
                            </option>
                            <option
                              value="0"
                            >
                              0
                            </option>
                          </select>
                        </label>
                      </div>
                      <span
                        class="rcx-box rcx-box--full rcx-css-4w7o7u"
                        id="react-aria-:ra:"
                      >
                        1
                      </span>
                      <i
                        aria-hidden="true"
                        class="rcx-box rcx-box--full rcx-icon--name-chevron-down rcx-icon rcx-css-6vi44e"
                      >
                        
                      </i>
                    </button>
                  </span>
                </div>
                <div
                  class="rcx-box rcx-box--full rcx-field rcx-css-1fuhgt0"
                >
                  <label
                    class="rcx-box rcx-box--full rcx-label rcx-box rcx-box--full rcx-field__label rcx-css-irxdij"
                    for=":r2:"
                  >
                    Max number of uses
                  </label>
                  <span
                    class="rcx-box rcx-box--full rcx-field__row"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-labelledby="react-aria-:ri:"
                      class="rcx-box rcx-box--full rcx-select rcx-css-1vw6rc6"
                      id=":r2:"
                      type="button"
                    >
                      <div
                        aria-hidden="true"
                        data-a11y-ignore="aria-hidden-focus"
                        data-react-aria-prevent-focus="true"
                        data-testid="hidden-select-container"
                        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
                      >
                        <label>
                          <select
                            name="maxUses"
                            tabindex="-1"
                          >
                            <option />
                            <option
                              value="5"
                            >
                              5
                            </option>
                            <option
                              value="10"
                            >
                              10
                            </option>
                            <option
                              value="25"
                            >
                              25
                            </option>
                            <option
                              value="50"
                            >
                              50
                            </option>
                            <option
                              value="100"
                            >
                              100
                            </option>
                            <option
                              value="0"
                            >
                              0
                            </option>
                          </select>
                        </label>
                      </div>
                      <span
                        class="rcx-box rcx-box--full rcx-css-4w7o7u"
                        id="react-aria-:ri:"
                      >
                        5
                      </span>
                      <i
                        aria-hidden="true"
                        class="rcx-box rcx-box--full rcx-icon--name-chevron-down rcx-icon rcx-css-6vi44e"
                      >
                        
                      </i>
                    </button>
                  </span>
                </div>
                <div
                  class="rcx-box rcx-box--full rcx-css-31obua"
                >
                  <button
                    class="rcx-box rcx-box--full rcx-button--primary rcx-button"
                    disabled=""
                    type="button"
                  >
                    <span
                      class="rcx-button--content"
                    >
                      Generate_New_Link
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="os-scrollbar os-scrollbar-horizontal os-theme-dark os-scrollbar-auto-hide os-scrollbar-auto-hide-hidden os-scrollbar-handle-interactive os-scrollbar-cornerless os-scrollbar-unusable"
            style="--os-scroll-percent: 0; --os-viewport-percent: 0; --os-scroll-direction: 0;"
          >
            <div
              class="os-scrollbar-track"
            >
              <div
                class="os-scrollbar-handle"
              />
            </div>
          </div>
          <div
            class="os-scrollbar os-scrollbar-vertical os-theme-dark os-scrollbar-auto-hide os-scrollbar-auto-hide-hidden os-scrollbar-handle-interactive os-scrollbar-cornerless os-scrollbar-unusable"
            style="--os-scroll-percent: 0; --os-viewport-percent: 0; --os-scroll-direction: 0;"
          >
            <div
              class="os-scrollbar-track"
            >
              <div
                class="os-scrollbar-handle"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders InviteError without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-vertical-bar rcx-css-ucmcg1"
    >
      <div
        class="rcx-box rcx-box--full rcx-css-ftwpdg"
      >
        <div
          class="rcx-box rcx-box--full rcx-css-1sl6k6j"
        >
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button rcx-css-x7bl3q rcx-css-1yzvz7u"
            title="Back"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-arrow-back rcx-icon rcx-css-4pvxx3"
            >
              ⦉
            </i>
          </button>
          <div
            class="rcx-box rcx-box--full rcx-css-x7bl3q rcx-css-1to6ka7"
            id="contextualbarTitle"
          >
            Invite users
          </div>
          <button
            aria-label="Close"
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button rcx-css-x7bl3q rcx-css-1yzvz7u"
            data-qa="ContextualbarActionClose"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
        </div>
      </div>
      <div
        class="rcx-box rcx-box--full rcx-css-1svuzur"
      >
        <div
          class="rcx-box rcx-box--full rcx-css-pln26h rcx-css-1cb6i7s"
          data-overlayscrollbars="host"
        >
          <div
            class="os-size-observer"
          >
            <div
              class="os-size-observer-listener"
            />
          </div>
          <div
            class=""
            data-overlayscrollbars-padding=""
            style="margin-right: 0px; margin-bottom: 0px; margin-left: 0px; top: 0px; left: 0px; width: calc(100% + 0px);"
          >
            <div
              class=""
              data-overlayscrollbars-viewport="overflowXHidden overflowYHidden"
              style="padding: 0px 0px 0px 0px;"
              tabindex="-1"
            >
              <div
                class="rcx-box rcx-box--full rcx-css-iag4sp"
              >
                <section
                  class="rcx-box rcx-box--full rcx-callout rcx-callout--danger rcx-css-1jggbrp rcx-css-1fuhgt0"
                >
                  <i
                    aria-hidden="true"
                    class="rcx-box rcx-box--full rcx-icon--name-ban rcx-icon rcx-callout__icon rcx-css-4pvxx3"
                  >
                    
                  </i>
                  <div
                    class="rcx-box rcx-box--full rcx-callout__wrapper"
                  >
                    <div
                      class="rcx-box rcx-box--full rcx-callout__wrapper-content"
                    >
                      <div
                        class="rcx-box rcx-box--full rcx-callout__content"
                      >
                        Error: Error message
                      </div>
                    </div>
                  </div>
                </section>
              </div>
            </div>
          </div>
          <div
            class="os-scrollbar os-scrollbar-horizontal os-theme-dark os-scrollbar-auto-hide os-scrollbar-auto-hide-hidden os-scrollbar-handle-interactive os-scrollbar-cornerless os-scrollbar-unusable"
            style="--os-scroll-percent: 0; --os-viewport-percent: 0; --os-scroll-direction: 0;"
          >
            <div
              class="os-scrollbar-track"
            >
              <div
                class="os-scrollbar-handle"
              />
            </div>
          </div>
          <div
            class="os-scrollbar os-scrollbar-vertical os-theme-dark os-scrollbar-auto-hide os-scrollbar-auto-hide-hidden os-scrollbar-handle-interactive os-scrollbar-cornerless os-scrollbar-unusable"
            style="--os-scroll-percent: 0; --os-viewport-percent: 0; --os-scroll-direction: 0;"
          >
            <div
              class="os-scrollbar-track"
            >
              <div
                class="os-scrollbar-handle"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders InviteLoading without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-vertical-bar rcx-css-ucmcg1"
    >
      <div
        class="rcx-box rcx-box--full rcx-css-ftwpdg"
      >
        <div
          class="rcx-box rcx-box--full rcx-css-1sl6k6j"
        >
          <button
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button rcx-css-x7bl3q rcx-css-1yzvz7u"
            title="Back"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-arrow-back rcx-icon rcx-css-4pvxx3"
            >
              ⦉
            </i>
          </button>
          <div
            class="rcx-box rcx-box--full rcx-css-x7bl3q rcx-css-1to6ka7"
            id="contextualbarTitle"
          >
            Invite users
          </div>
          <button
            aria-label="Close"
            class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button rcx-css-x7bl3q rcx-css-1yzvz7u"
            data-qa="ContextualbarActionClose"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
            >
              
            </i>
          </button>
        </div>
      </div>
      <div
        class="rcx-box rcx-box--full rcx-css-1svuzur"
      >
        <div
          class="rcx-box rcx-box--full rcx-css-pln26h rcx-css-1cb6i7s"
          data-overlayscrollbars="host"
        >
          <div
            class="os-size-observer"
          >
            <div
              class="os-size-observer-listener"
            />
          </div>
          <div
            class=""
            data-overlayscrollbars-padding=""
            style="margin-right: 0px; margin-bottom: 0px; margin-left: 0px; top: 0px; left: 0px; width: calc(100% + 0px);"
          >
            <div
              class=""
              data-overlayscrollbars-viewport="overflowXHidden overflowYHidden"
              style="padding: 0px 0px 0px 0px;"
              tabindex="-1"
            >
              <div
                class="rcx-box rcx-box--full rcx-css-iag4sp"
              >
                <span
                  class="rcx-skeleton rcx-skeleton--text rcx-css-1jggbrp rcx-css-1qcz93u"
                />
              </div>
            </div>
          </div>
          <div
            class="os-scrollbar os-scrollbar-horizontal os-theme-dark os-scrollbar-auto-hide os-scrollbar-auto-hide-hidden os-scrollbar-handle-interactive os-scrollbar-cornerless os-scrollbar-unusable"
            style="--os-scroll-percent: 0; --os-viewport-percent: 0; --os-scroll-direction: 0;"
          >
            <div
              class="os-scrollbar-track"
            >
              <div
                class="os-scrollbar-handle"
              />
            </div>
          </div>
          <div
            class="os-scrollbar os-scrollbar-vertical os-theme-dark os-scrollbar-auto-hide os-scrollbar-auto-hide-hidden os-scrollbar-handle-interactive os-scrollbar-cornerless os-scrollbar-unusable"
            style="--os-scroll-percent: 0; --os-viewport-percent: 0; --os-scroll-direction: 0;"
          >
            <div
              class="os-scrollbar-track"
            >
              <div
                class="os-scrollbar-handle"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
